kotest-framework/kotest-framework-concurrency/src/commonMain/kotlin/io/kotest/framework/concurrency/continually.kt

```diff
@@ -38,28 +47,34 @@ class ContinuallyBuilder<T> {
 }
 
 @Deprecated("Replaced with the io.kotest.assertions.nondeterministic utils. Deprecated in 5.7")
-data class ContinuallyState<T>(val result: T, val start: Long, val end: Long, val times: Int)
+data class ContinuallyState<T>(
+   val result: T,
+   val start: Long,
+   val end: Long,
+   val times: Int,
+)
 
 @Deprecated("Replaced with the io.kotest.assertions.nondeterministic utils. Deprecated in 5.7")
 private suspend fun <T> ContinuallyConfig<T>.invoke(f: suspend () -> T): T? {
    delay(initialDelay)
 
-   val start = timeInMillis()
-   val end = start + duration
+   val duration = duration.milliseconds
+   val start = TimeSource.Monotonic.markNow()
+   val end = start.plus(duration)
    var times = 0
    var result: T? = null
 
-   while (timeInMillis() < end) {
+   while (end.hasNotPassedNow()) {
       try {
          result = f()
-         listener?.invoke(ContinuallyState(result, start, end, times))
+         listener?.invoke(ContinuallyState(result, 0L, duration.inWholeMilliseconds, times))
```

Comment:
Just to highlight, there's a potentially breaking change here. 

The start and end of a ContinuallyState will no longer reflect the system time, but start will always be 0, and end will always be the configured duration.

I think this change is acceptable because it's a minor change, and this code has been deprecated for 3+ years.

Reply:
agreed, all deprecated functions are coming out for 6.0